plan/plan.go

```diff
@@ -275,6 +275,7 @@ type Rel interface {
 	// record type from the output RecordType of the underlying Relation.
 	// If the output mapping is nil, then this just returns the input
 	// type.
+	// TODO - Rename this ApplyMap.
```

Comment:
Add api to interface.

Also, I think we should be copying, not overwriting. Overwriting/mutating nodes in a plan can be dangerous given the possibility of duplicated subplans, etc.

Maybe call CopyWithRemap() or similar.

Reply:
Remap doesn't really make sense on write relations so I specifically created an alternative interface.

I'm less inclined to worry about copying as these are more of a build time artifact (i.e. not protos).  If one wants to duplicate a subplan that should do the copying.